### ISSUE TITLE
Add/Edit SmartStart provisioning list

### ISSUE THREAD

**Issue by marcelveldt** [zwave-js/backlog#11]
From core created by [AlCalzone](https://github.com/AlCalzone): home-assistant/core#94067  

### The problem

**A controller providing control of the SmartStart functionality MUST:**
- [x] Provide a method for the end user to view the Node Provisioning List entries with their network status (included/ not included or failed) --> _double check!_
- [x] Provide a method for the end user to manually add and remove entries in the Node Provisioning List.
- [ ] Provide a method for the end user to edit available settings for each entry in the Node Provisioning List. (e.g., Inclusion setting, Advanced joining)
- [x] Support S2 inclusion with authentication using the DSK PIN code.

- [x] If a user removes a node from the Node Provisioning List and the node is still included in the Z-Wave network, the controller MUST inform the end user that the node will stay in the network and requires to be excluded manually or reset to factory default in order to leave the Z-Wave Network.

- [x] A controller MUST inform the end user that S2 only (non-SmartStart) nodes present in the Provisioning List require to perform a classic inclusion to add them into the Z-Wave network. __Problem avoided, see comments below__

Currently there's no such UI in Home Assistant.

- [ ] When done, this also needs to be added to the documentation/manual.

Advanced joining = selecting which keys shall be granted (optional)

### What version of Home Assistant Core has the issue?

core-2023.7.0.dev20230605

### What type of installation are you running?

Home Assistant OS

**Comment by marcelveldt** [comment]
Hey there @home-assistant/z-wave, mind taking a look at this issue as it has been labeled with an integration (`zwave_js`) you are listed as a [code owner](https://github.com/home-assistant/core/blob/dev/CODEOWNERS#L1446) for? Thanks!

<details>
<summary>Code owner commands</summary>


Code owners of `zwave_js` can trigger bot actions by commenting:

- `@home-assistant close` Closes the issue.
- `@home-assistant rename Awesome new title` Renames the issue.
- `@home-assistant reopen` Reopen the issue.
- `@home-assistant unassign zwave_js` Removes the current integration label and assignees on the issue, add the integration domain after the command.

</details>


<sub><sup>(message by CodeOwnersMention)</sup></sub>

---

[zwave_js documentation](https://www.home-assistant.io/integrations/zwave_js)
[zwave_js source](https://github.com/home-assistant/core/tree/dev/homeassistant/components/zwave_js)
<sub><sup>(message by IssueLinks)</sup></sub>

**Comment by marcelveldt** [comment]
We have the first bullet already I believe.

**Comment by marcelveldt** [comment]
> Why would there be non SmartStart devices in the provisoning list? That sounds useless.

I don't think this makes sense, but we have already solved it I think?
When you scan a QR code that belongs to a non-smartstart device, it instructs you to do a normal inclusion. At least I remember talking about this with Raman.

**Comment by marcelveldt** [comment]
Yeah, I don't think we allow you to provision a non SmartStart device.

**Comment by marcelveldt** [comment]
> If you had at least one provisioned device there used to be a link to a list in the Z-Wave configuration panel.

still there it looks like:
- https://github.com/home-assistant/frontend/blob/dev/src/panels/config/integrations/integration-panels/zwave_js/zwave_js-config-dashboard.ts#L232-L241
- https://github.com/home-assistant/frontend/blob/dev/src/panels/config/integrations/integration-panels/zwave_js/zwave_js-provisioned.ts

I see in the second bullet that we have included and not included, I don't see anything about failed though. How do we know it failed?

**Comment by marcelveldt** [comment]
for the second to last bullet, this is what we say when you remove a provisioning entry, is this sufficient?

"If you unprovision the device it will not be added to Home Assistant when it is powered on. If it is already added to Home Assistant, removing the provisioned device will not remove it from Home Assistant."

It would be straight forward to have specific text based on whether the device is included or not if that's a specific requirement

**Comment by marcelveldt** [comment]
> I don't see anything about failed though. How do we know it failed?

The tests seem to use this in situations where the driver considers a node **dead**. The tests didn't check this though, so maybe it's safe not to put the status there.

**Comment by marcelveldt** [comment]
The original issue from core (where this issue is migrated from) had a few comments. The gist is that we do not need any new functionality to add non smartstart devices as the Ui already points the user to do a regular inclusion.

**Comment by AlCalzone** [comment]
It seems that most of this is done. I'm not 100% sure that this is also solved
> Provide a method for the end user to edit available settings for each entry in the Node Provisioning
List. (e.g., Inclusion setting, Advanced joining). A controller application MAY provide
no available settings.

...since there is no way to edit the settings, but the certification test still passed although I selected it is not possible. Would be nice to have though.

**Comment by MartinHjelmare** [comment]
What's the status here?

**Comment by AlCalzone** [comment]
It would be nice to be able to edit the existing provisioning entries - deselecting security classes, possibly switching from Mesh to LR, predefining name and location (?).
Not required for certification though.

**Comment by Dlivera** [comment]
I have the same issue with new device stuck in provisioning.
